pkg/controller/utils/component.go

```diff
@@ -449,6 +449,20 @@ func mergeState(desired client.Object, current runtime.Object) client.Object {
 		// APIServer.
 		dui.SetOwnerReferences(cui.GetOwnerReferences())
 		return dui
+	case *v3.NetworkPolicy:
+		cnp := current.(*v3.NetworkPolicy)
+		dnp := desired.(*v3.NetworkPolicy)
+		if reflect.DeepEqual(cnp.Spec, dnp.Spec) {
+			return nil
+		}
+		return dnp
+	case *v3.Tier:
+		ct := current.(*v3.Tier)
+		dt := desired.(*v3.Tier)
+		if reflect.DeepEqual(ct.Spec, dt.Spec) {
+			return nil
```

Comment:
Other resources in this file return the current object when they don't want a change. I assume we should change those to nil?

Didn't want to bring that in discussion/testing in scope here though.